aws_sagemaker_studio/sagemaker_studio_image_build/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,740 @@
+    "image_tag = ecr.list_images(repositoryName=domain_id, filter={'tagStatus':'TAGGED'})['imageIds'][0]['imageTag']\n",
+    "ecr_image = '{}.dkr.ecr.{}.amazonaws.com/{}:{}'.format(account, region, domain_id, image_tag)\n",
+    "\n",
+    "print(ecr_image)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train our model using SageMaker Training Instances"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.estimator import Estimator\n",
+    "\n",
+    "hyperparameters = {'train-steps': 100}\n",
+    "\n",
+    "instance_type = 'ml.m4.xlarge'\n",
+    "\n",
+    "estimator = Estimator(role=role,\n",
+    "                      train_instance_count=1,\n",
+    "                      train_instance_type=instance_type,\n",
+    "                      image_name=ecr_image,\n",
+    "                      hyperparameters=hyperparameters)\n",
+    "\n",
+    "estimator.fit(data_location)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Host our model using SageMaker Hosting"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = estimator.deploy(1, instance_type)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Making predictions using Python SDK\n",
+    "To make predictions, we use an image that is converted using OpenCV into a json format to send as an inference request. We need to install OpenCV to deserialize the image that is used to make predictions.\n",
+    "\n",
+    "The JSON reponse will be the probabilities of the image belonging to one of the 10 classes along with the most likely class the picture belongs to. The classes can be referenced from the CIFAR-10 website. Since we didn't train the model for that long, we aren't expecting very accurate results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!apt-get update\n",
+    "!apt-get install -y libsm6 libxext6 libxrender-dev libgl1-mesa-glx"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install opencv-python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "import numpy\n",
+    "\n",
+    "from sagemaker.predictor import json_serializer, json_deserializer\n",
+    "\n",
+    "# Update the image path below based on the path to the /data/cat.png image on Studio\n",
+    "image = cv2.imread(\"data/cat.png\", 1)\n",
+    "\n",
+    "# resize, as our model is expecting images in 32x32.\n",
+    "image = cv2.resize(image, (32, 32))\n",
```

Comment:
If we're doing TF, why not use tf, or just use something already installed in the data science kernel to do a resize.

Reply:
This is based on adapting existing sagemaker notebook examples in advanced functionality (notebook instance) to sagemaker studio.   There are dependencies that are not included in Studio that the existing notebook relies on.  I can refactor the existing but we were intending to show adapting existing BYO notebooks.